src/lib/bakeryjs/tracingModel.ts

```diff
@@ -435,4 +414,41 @@ export class TracingModel {
 		);
 		return;
 	}
+
+	private getSubDimensionsDone(msgId: string) {
+		if (!this.dimensionStore.has(msgId)) return true;
+
+		const dimValues = this.dimensionStore.get(msgId).values();
+		return every(
+			[...dimValues],
```

Comment:
no need to have this spreading since `every` should not mutate the array.

Reply:
This is not an array. It's the result of `.values()` from a Map. You must convert it to any array.

Reply:
Hmmmm, got it.
So could you benchmark a custom `.every` for the Map type?

Something like:

```javascript
for (let value of a.values()){
	if (!evalFunction(value)) {
       return false
    }
}
return true
```